### PR TITLE
Return tv episode object as JSON

### PR DESCRIPTION
**Pull request by fernandog**
@OmgImAlexis @ratoaq2 
I tried:

```python
    def to_json(self):
        """JSON representation."""
        return json.dumps(self)
```

but i get this:
```python
  File "/usr/lib/python2.7/json/encoder.py", line 184, in default
    raise TypeError(repr(o) + " is not JSON serializable")
TypeError: <medusa.tv.TVEpisode object at 0x6cf69fb0> is not JSON serializable
```

maybe because tv_episode.show is a TV Show class?

### CODE REVIEW

**Comment by ratoaq2** [comment]
https://github.com/ratoaq2/knowit/blob/master/knowit/utils.py#L4

**Comment by fernandog** [comment]
@ratoaq2 

```python
  File "/usr/lib/python2.7/json/encoder.py", line 184, in default
    raise TypeError(repr(o) + " is not JSON serializable")
TypeError: <thread.lock object at 0x7197b5a0> is not JSON serializable

```

**Comment by OmgImAlexis** [comment]
@fernandog should a test be added for this?

**Comment by fernandog** [comment]
@OmgImAlexis its not working. waiting @ratoaq2 to help see what's wrong

**Comment by fernandog** [comment]
still:
  File "/usr/lib/python2.7/json/encoder.py", line 184, in default
    raise TypeError(repr(o) + " is not JSON serializable")
TypeError: <thread.lock object at 0x716e17d0> is not JSON serializable

**Comment by OmgImAlexis** [comment]
@fernandog why was this closed?

**Comment by OmgImAlexis** [comment]
I'll test this to make sure there's no crashes and then we can merge.

**Comment by OmgImAlexis** [comment]
Here's an example output.

```
{
  "show": {
    "name": "Lucifer",
    "indexer": 1,
    "indexerId": 295685,
    "network": "FOX (US)",
    "genres": [
      "Comedy",
      "Drama",
      "Fantasy"
    ],
    "classification": "Scripted",
    "runtime": "45",
    "imdbId": "tt4052886",
    "imdbInfo": {
      "rating": 8.3,
      "votes": 56781,
      "runtimes": "42",
      "year": 2015,
      "imdb_id": "tt4052886",
      "akas": "Εωσφόρος::Greece|Луцифер::Bulgaria (Bulgarian title)|Люцифер::Ukraine|Люцифер::Russia|Liuciferis::Lithuania (imdb display title)|Lucifer::Spain (imdb display title)|Lucifer::Hungary (imdb display title)|Lucyfer::Poland (imdb display title)",
      "genres": "Crime|Drama|Fantasy",
      "countries": "USA",
      "title": "Lucifer",
      "last_update": 736259,
      "country_codes": "us",
      "certificates": "TV-14"
    },
    "quality": 336,
    "flattenFolders": false,
    "status": "Continuing",
    "airs": "Monday 9:00 PM",
    "startYear": 2016,
    "paused": false,
    "airByDate": false,
    "subtitlesEnabled": false,
    "dvdOrder": 0,
    "lang": "en",
    "lastUpdateIndexer": 736259,
    "sports": false,
    "anime": false,
    "scene": false,
    "nextAired": "",
    "location": "/Users/xo/fakeShows/Lucifer",
    "releaseGroups": null,
    "defaultEpisodeStatus": 5,
    "releaseIgnoredWords": [],
    "releaseRequiredWords": [],
    "exceptions": []
  }
}
```

**Comment by fernandog** [comment]
its this ok to merge? need approve

**Comment by OmgImAlexis** [comment]
@fernandog still want to change a few things before merging.

**Comment by OmgImAlexis** [comment]
@ratoaq2 can you please rebase this and have a look at my changes.

File: medusa/tv.py [hunk 1]
```diff
@@ -1478,6 +1479,63 @@ def __unicode__(self):
         to_return += u'anime: {0}\n'.format(self.is_anime)
         return to_return
 
+    def to_json(self):
+        """Return JSON representation."""
+        # @TODO: Should we also return the season list with all episodes?
+        genres = list(set([v for v in self.genre.split('|') if v] + [v for v in self.imdb_info['genres'].replace('Sci-Fi', 'Science-Fiction').split('|') if v]))
+        indexers = {
```
Comment (p0psicles) [inline medusa/tv.py]: Is it really needed to pass indexer configuration here? I'd rather keep it in one place?

Comment (OmgImAlexis) [inline medusa/tv.py]: It'd be great if we had something like `indexer.find_by_id(1)` or `indexer.find_by_name('thetvdb')`and it'd return the indexer.

Comment (p0psicles) [inline medusa/tv.py]: The indexers config located in medusa/indexers/indexer_config.py.
You could from anywhere do this:
from medusa.indexers.indexer_config import indexerConfig
indexers = {x:y['name'] for x,y in indexerConfig.iteritems()}
```dict: {1: 'TVDBv2'}```

Comment (OmgImAlexis) [inline medusa/tv.py]: Okay. I'll look into using that.

**Comment by fernandog** [comment]
@OmgImAlexis what else is needed? can we merge and you continue in your other PR?

**Comment by fernandog** [comment]
@OmgImAlexis what do you think about removing your last commit and merge this? so you can work in your local branch and we keep the scope of the PR?

**Comment by OmgImAlexis** [comment]
@fernandog that's part of the PR's scope. There's no point in merging this and then having yet another PR to just change the names of the fields, etc.

If @ratoaq2 thinks this is okay I say just merge as is once it's been rebased and we should be okay. I don't really want to rename much more at the moment and won't have full internet access for another few days.

**Pull request merged**